tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -107,6 +108,14 @@ class CGMSHLSLRuntime : public CGHLSLRuntime {
                  llvm::SmallVector<std::pair<DXIL::ResourceClass, unsigned>, 1>>
       constantRegBindingMap;
 
+  // Map from type to resource properties.
+  // This only has info about type, globalcoherent is saved in
+  // globalCoherentSet.
+  DenseMap<llvm::Type *, DxilResourceProperties> typeToResPropertiesMap;
```

Comment:
This is insufficient to pick up all the resource properties, such as unorm component type (not in struct with type annotation).  Maybe we have the necessary info at the site of the cast from Resource.